master/internal/db/postgres_users.go

```diff
@@ -43,7 +43,12 @@ func (db *PgDB) StartUserSession(user *model.User) (string, error) {
 }
 
 // UserByToken returns a user session given an authentication token.
-func (db *PgDB) UserByToken(token string) (*model.User, *model.UserSession, error) {
+func (db *PgDB) UserByToken(token string, ext *model.ExternalSessions) (
```

Comment:
Meta comment: it would be great if this could be served by a generic plugin, in which case we should pass the whole config. But that was impractical for other reason and this is sufficiently extensible for now.